### ISSUE TITLE
Set up Better Sharing unit testing

### ISSUE THREAD

**Issue by dangerouse** [cloudsponge/better-sharing-wp#88]
Let's ensure that our plugin is able to run automated unit tests. Please set up the framework and commands needed to provide some basic coverage of our core modules.

**Comment by royboy789** [comment]
We discussed this internally today.

@dangerouse - we will put in some PHP unit tests. I don't think we need JS unit tests, however we are looking into better user testing tools to automate testing for every version. Something like ghostinspector.com/

**Comment by dangerouse** [comment]
@royboy789 Ghost Inspector looks interesting. It appears to be an integration testing product and I think there is definitely a place for something like this down the line. 

For this issue, I'd like to address unit testing so that we can test each module in isolation. To be clear, I'm not wanting to create a bunch of tests to cover the code so far. I just want us to have a unit testing framework in place so that we can add test cases as we uncover bugs. 

It looks like GH actions [point towards using Composer](https://github.com/actions/starter-workflows/blob/804289e1b5bf70f30e27dfe2c8c691c8cddd384a/ci/php.yml). What do you think about is as an option?

**Comment by royboy789** [comment]
@dangerouse yes, for PHP we would use composer and [PHPUnit](https://phpunit.de/) for unit testing and test coverage.
This won't cover the entirety of the user experience per component w/o JavaScript or user testing, but will be a good place to start.
